packages/@winglang/sdk/src/target-sim/bucket.inflight.ts

```diff
@@ -398,3 +594,26 @@ export class Bucket implements IBucketClient, ISimulatorResourceInstance {
     });
   }
 }
+
+async function isPortAvailable(port: number): Promise<boolean> {
+  return new Promise((resolve, _reject) => {
+    const s = new Socket();
+    s.once("error", (err) => {
+      s.destroy();
+      if ((err as any).code !== "ECONNREFUSED") {
+        resolve(false);
+      } else {
+        // connection refused means the port is not used
+        resolve(true);
+      }
+    });
+
+    s.once("connect", () => {
+      s.destroy();
+      // connection successful means the port is used
+      resolve(false);
+    });
+
+    s.connect(port, LOCALHOST_ADDRESS);
+  });
+}
```

Comment:
I think this utility function is reused elsewhere, can we put it in a shared file?